app/src/main/java/org/ramonaza/officialramonapp/people/rides/backend/RidesOptimizer.java

```diff
@@ -13,21 +14,27 @@
 public class RidesOptimizer {
 
     /**
+     * Calculate rides based on latitude and longitude, finding the optimal assignment of passengers
+     * to drivers in order to minimize total distance traveled by all drivers. Makes the simplifying
+     * assumption that all drivers return to their home in between each drop-off.
+     */
+    public static final int ALGORITHM_NAIVE_HUNGARIAN=0;
+    /**
      * Calculate rides based on latitude and longitude, iterating over the passengers
      * and assigning them a driver.
-      */
-    public static final int ALGORITHM_LATLONG_ALEPHS_FIRST=0;
+     */
+    public static final int ALGORITHM_LATLONG_ALEPHS_FIRST=1;
     /**
      * Calculate rides based on latitude and longitude, iterating over the drivers
      * and assigning them passengers.
      */
-    public static final int ALGORITHM_LATLONG_DRIVERS_FIRST=1;
-    private Set<ContactInfoWrapper> alephsToOptimize;
+    public static final int ALGORITHM_LATLONG_DRIVERS_FIRST=2;
+    private List<ContactInfoWrapper> alephsToOptimize;
     private List<DriverInfoWrapper> driversToOptimize;
     private int algorithm;
     private boolean retainPreexisting;
     public RidesOptimizer(){
-        this.alephsToOptimize=new HashSet<ContactInfoWrapper>();
+        this.alephsToOptimize=new ArrayList<ContactInfoWrapper>();
```

Comment:
It was necessary for me to be able to associate indices with the elements. I did notice the issue with duplicates. If this is the cause, I can fix it when I have access to my desktop again.